planner/cascades/testdata/integration_suite_out.json

```diff
@@ -163,52 +138,41 @@
-          "  └─Projection_10 10000.00 root cast(Column#1)",
-          "    └─TableReader_8 10000.00 root data:TableScan_9",
-          "      └─TableScan_9 10000.00 cop[tikv] table:t, range:[-inf,+inf], keep order:false, stats:pseudo"
-        ]
-      },
-      {
-        "SQL": "select a from t",
-        "Result": [
-          "1",
-          "2",
-          "3",
-          "4"
-        ]
-      },
-      {
-        "SQL": "explain select max(a), min(b) from t",
+        "SQL": "select sum(a) from t",
+        "Plan": [
+          "Projection_8 1.00 root Column#3",
+          "└─HashAgg_13 1.00 root funcs:sum(Column#5)",
+          "  └─TableReader_14 1.00 root data:HashAgg_15",
+          "    └─HashAgg_15 1.00 cop[tikv] funcs:sum(Column#1)",
+          "      └─TableScan_11 10000.00 cop[tikv] table:t, range:[-inf,+inf], keep order:false, stats:pseudo"
+        ],
         "Result": [
-          "Projection_6 1.00 root Column#3, Column#4",
-          "└─HashAgg_7 1.00 root funcs:max(Column#1), min(Column#2)",
-          "  └─TableReader_8 10000.00 root data:TableScan_9",
-          "    └─TableScan_9 10000.00 cop[tikv] table:t, range:[-inf,+inf], keep order:false, stats:pseudo"
+          "10"
         ]
       },
       {
         "SQL": "select max(a), min(b) from t",
+        "Plan": [
+          "Projection_8 1.00 root Column#3, Column#4",
+          "└─HashAgg_12 1.00 root funcs:max(Column#7), min(Column#8)",
+          "  └─TableReader_13 1.00 root data:HashAgg_14",
+          "    └─HashAgg_14 1.00 cop[tikv] funcs:max(Column#1), min(Column#2)",
+          "      └─TableScan_11 10000.00 cop[tikv] table:t, range:[-inf,+inf], keep order:false, stats:pseudo"
+        ],
         "Result": [
           "4 11"
         ]
       },
-      {
-        "SQL": "explain select b, avg(a) from t group by b order by b",
-        "Result": [
-          "Projection_8 8000.00 root Column#2, Column#3",
-          "└─Sort_14 8000.00 root Column#2:asc",
-          "  └─HashAgg_10 8000.00 root group by:Column#8, funcs:avg(Column#6), firstrow(Column#7)",
-          "    └─Projection_13 10000.00 root cast(Column#1), Column#2, Column#2",
-          "      └─TableReader_11 10000.00 root data:TableScan_12",
-          "        └─TableScan_12 10000.00 cop[tikv] table:t, range:[-inf,+inf], keep order:false, stats:pseudo"
-        ]
-      },
       {
         "SQL": "select b, avg(a) from t group by b order by b",
+        "Plan": [
+          "Projection_10 8000.00 root Column#2, Column#3",
+          "└─Sort_19 8000.00 root Column#2:asc",
+          "  └─HashAgg_12 8000.00 root group by:Column#12, funcs:avg(Column#10), firstrow(Column#11)",
+          "    └─Projection_15 10000.00 root cast(Column#1), Column#2, Column#2",
+          "      └─TableReader_13 10000.00 root data:TableScan_14",
+          "        └─TableScan_14 10000.00 cop[tikv] table:t, range:[-inf,+inf], keep order:false, stats:pseudo"
+        ],
```

Comment:
I have tried this query in the old planner, it doesn't push this agg down, either.
```
mysql> create table t (a int primary key, b int);
Query OK, 0 rows affected (0.01 sec)

mysql> explain select b, avg(a) from t group by b order by b;
+----------------------------+----------+-----------+---------------------------------------------------------------+
| id                         | count    | task      | operator info                                                 |
+----------------------------+----------+-----------+---------------------------------------------------------------+
| Sort_5                     | 8000.00  | root      | Column#4:asc                                                  |
| └─Projection_7             | 8000.00  | root      | Column#2, Column#3                                            |
|   └─HashAgg_10             | 8000.00  | root      | group by:Column#12, funcs:avg(Column#10), firstrow(Column#11) |
|     └─Projection_17        | 10000.00 | root      | cast(Column#1), Column#2, Column#2                            |
|       └─TableReader_15     | 10000.00 | root      | data:TableScan_14                                             |
|         └─TableScan_14     | 10000.00 | cop[tikv] | table:t, range:[-inf,+inf], keep order:false, stats:pseudo    |
+----------------------------+----------+-----------+---------------------------------------------------------------+
6 rows in set (0.01 sec)
```